test/solidityExecutionFramework.h

```diff
@@ -46,7 +46,20 @@ class ExecutionFramework
 	bytes const& compileAndRun(std::string const& _sourceCode, u256 const& _value = 0, std::string const& _contractName = "")
 	{
 		dev::solidity::CompilerStack compiler;
-		compiler.compile(_sourceCode, m_optimize);
+		try
+		{
+			compiler.compile(_sourceCode, m_optimize);
+		}
+		catch (const std::exception& e)
+		{
+			std::string const* extra = boost::get_error_info<errinfo_comment>(e);
+			std::string msg = std::string("Parsing contract failed with: ") +
+				e.what() + std::string("\n");
```

Comment:
you could just do "msg =std::string("Parsing contract failed with: ") + diagnostic_information(_e);" Then you get all possible information.